directory/cli/src/user.py

```diff
@@ -74,7 +74,93 @@ def show(login):
             error = '{}: user not found'.format(login)
             raise click.ClickException(error)
 
-    wrapper_commands = [
+    if not utils.advanced_mode_enabled():
+        @user.command(help='Create a new user')
+        def create():
+            click.echo('Please enter details for the following:')
+
+            params = {
+                        'login': click.prompt('  Username'),
+                        'first': click.prompt('  First name'),
+                        'last': click.prompt('  Surname'),
+                        'email': click.prompt('  Email'),
+            }
+            if click.confirm('Would you like to add a UID to this user?'):
+                params = { **params, 'uid': click.prompt('  UID') }
+
+            group_id = _get_group_id('clusterusers')
+
+            if group_id:
+                params = { **params, 'gidnumber': group_id[0] }
+
+            wrapper = ipa_wrapper_command.create_ipa_wrapper(
+                'user-add',
+                argument_name='login',
+                transform_options_callback=_transform_create_options,
+                handle_result_callback=_handle_create_result,
+            )
+            wrapper(**params)
+
+        @user.command(help='Modify an existing user')
+        def modify():
+            click.echo('Please enter the name of the user you want to modify:')
+            user = click.prompt('  Username')
+            user_find_args = ['--login={}'.format(user)]
+            try:
+                user_data = list_command.find_data(
+                        'user-find',
+                        user_find_args,
+                        all_fields=True
+                )[0]
+            except IpaRunError:
+                # Matches error shown for user show, can extract logic to a method
+                # in the future
+                error = '{}: user not found'.format(user)
+                raise click.ClickException(error)
+
+            click.echo('Modifying user: %s' % user)
+            params = {
+                'login': user,
+                'first': click.prompt(
+                    '  First name',
+                    default=user_data.get('First name', 'empty')[0]
+                ),
+                'last': click.prompt(
+                    '  Surname',
+                    default=user_data.get('Last name', 'empty')[0]
+                ),
+                'email': click.prompt(
+                    '  Email',
+                    default=user_data.get('Email address', 'empty')[0]
+                )
+            }
+
+            wrapper = ipa_wrapper_command.create_ipa_wrapper(
+                'user-mod',
+                argument_name='login',
+                transform_options_callback=_transform_modify_options,
+                handle_result_callback=_handle_modify_result,
+            )
+            wrapper(**params)
+
+    base_ipa_commands = [
```

Comment:
Not sure about the naming of this and `advanced_ipa_commands` - these arrays contain the `directory user` commands, which wrap IPA commands but aren't themselves IPA commands, and are now potentially not the only commands since we define the simple mode commands above. Therefore I think it would be clearer to name these `base_ipa_wrapper_commands` and `advanced_ipa_wrapper_commands`.